internal/controller/bucket/bucket.go

```diff
@@ -290,6 +291,28 @@ func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 	if !ok {
 		return managed.ExternalUpdate{}, errors.New(errNotBucket)
 	}
+
+	c.log.Info("Updating bucket on all available s3 backends", "bucket name", bucket.Name)
+
+	g := new(errgroup.Group)
+	for _, s3Backend := range c.backendStore.GetAllBackends() {
+		backend := s3Backend
+		g.Go(func() error {
+			var err error
+			for i := 0; i < requestRetries; i++ {
+				err = c.update(ctx, bucket, backend)
```

Comment:
What would happen if some of the backends are able to update while other not? During create it isn't a "big" drama, because we have the same data, just with less replicas, but with update we should find our self in an inconsistent state. WDYT?

Reply:
I think there are two options:
1. Only perform the update if **_all_** buckets can be updated successfully.
2. If a particular backend is unable to update, then we update that backend's `ProviderConfg` status as `Unhealthy` (as per https://github.com/linode/provider-ceph/pull/11) and carry on.

I prefer option 2, as 1 is a bit messy and creates a single point of failure.

Reply:
On case 1. we have to do some revert or something "transaction" like
On case 2. It sounds great, but in my understanding we don't want to fetch bucket metadata from Kubernetes api server each time, so how other systems would know a bucket on a specific node is up to date or not?

Reply:
How about a map in the bucket status of backend name and whether the bucket is up to date on each backend?

eg
```yaml
kind: Bucket
name: my-bucket
status:
  syncedOnBackend:
    ceph-cluster-1: true
    ceph-cluster-2: false
    ceph-cluster-3: true
```